zokrates_core/src/compile.rs

```diff
@@ -25,19 +24,26 @@ use zokrates_field::Field;
 use zokrates_pest_ast as pest;
 
 #[derive(Debug)]
-pub struct CompilationArtifacts<T: Field> {
-    prog: ir::Prog<T>,
-    abi: Abi,
+pub struct CompilationArtifacts<T, I: IntoIterator<Item = ir::Statement<T>>> {
+    pub prog: ir::ProgIterator<T, I>,
+    pub abi: Abi,
```

Comment:
is there a reason for this to be public? `abi` function (L37) is implemented for this struct which returns a reference

Reply:
Fixed with `into_inner`, the existing accessors were not sufficient because of borrowing rules.